DB/src/main/java/io/deephaven/db/v2/sources/chunkcolumnsource/CharChunkColumnSource.java

```diff
@@ -0,0 +1,133 @@
+import io.deephaven.util.QueryConstants;
+import org.apache.commons.lang3.mutable.MutableInt;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+
+public class CharChunkColumnSource extends AbstractColumnSource<Character> implements ImmutableColumnSourceGetDefaults.ForChar, ChunkColumnSource<Character> {
+    private final ArrayList<CharChunk<? extends Attributes.Values>> data = new ArrayList<>();
+    private final TLongArrayList offsets = new TLongArrayList();
+    private long totalSize = 0;
+
+    // region constructor
+    protected CharChunkColumnSource() {
+        super(Character.class);
+    }
+    // endregion constructor
+
+    @Override
+    public char getChar(long index) {
+        if (index < 0 || index >= totalSize) {
+            return QueryConstants.NULL_CHAR;
+        }
+
+        final int chunkIndex = getChunkIndex(index);
+        final long offset = offsets.getQuick(chunkIndex);
+        return data.get(chunkIndex).get((int)(index - offset));
+    }
+
+    private final static class ChunkGetContext<ATTR extends Attributes.Any> extends DefaultGetContext<ATTR> {
+        private final ResettableCharChunk resettableCharChunk = ResettableCharChunk.makeResettableChunk();
+
+        public ChunkGetContext(ChunkSource<ATTR> chunkSource, int chunkCapacity, SharedContext sharedContext) {
+            super(chunkSource, chunkCapacity, sharedContext);
+        }
+
+        @Override
+        public void close() {
+            resettableCharChunk.close();
+        }
+    }
+
+    @Override
+    public GetContext makeGetContext(int chunkCapacity, SharedContext sharedContext) {
+        return new ChunkGetContext(this, chunkCapacity, sharedContext);
+    }
+
+    @Override
+    public Chunk<? extends Attributes.Values> getChunk(@NotNull GetContext context, @NotNull OrderedKeys orderedKeys) {
+        // if we can slice part of one of our backing chunks, then we will return that instead
+        if (orderedKeys.isContiguous()) {
+            final long firstKey = orderedKeys.firstKey();
+            final int firstChunk = getChunkIndex(firstKey);
+            final int lastChunk = getChunkIndex(orderedKeys.lastKey(), firstChunk);
+            if (firstChunk == lastChunk) {
+                final int offset = (int)(firstKey - offsets.get(firstChunk));
+                final int length = orderedKeys.intSize();
+                final CharChunk<? extends Attributes.Values> charChunk = data.get(firstChunk);
+                if (offset == 0 && length == charChunk.size()) {
+                    return charChunk;
+                }
+                return ((ChunkGetContext)context).resettableCharChunk.resetFromChunk(charChunk, offset, length);
+            }
+        }
+        return getChunkByFilling(context, orderedKeys);
+    }
+
+    @Override
+    public void fillChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Attributes.Values> destination, @NotNull OrderedKeys orderedKeys) {
```

Comment:
I'm kind of suspicious that you should just implement `com.illumon.iris.db.v2.sources.chunk.page.PageStore` and introduce a new kind of `ChunkPage` that has-a chunk instead of the current impl that is-a chunk.

That said, I think this code is perfectly correct barring other comments.